mne/encoding/feature.py

```diff
@@ -0,0 +1,298 @@
+        if event_ixs.ndim > 1:
+            raise ValueError("events must be shape (n_events,),"
+                             " found shape %s" % str(event_ixs.shape))
+        event_ids = np.ones_like(event_ixs) if event_ids is None else event_ids
+        unique_event_types = np.unique(event_ids)
+        covs, cov_names = _check_covariates(event_ixs, covariates,
+                                            covariate_names)
+        n_covs = covs.shape[-1]
+
+        # Create names for event types
+        if event_dict is None:
+            event_dict = dict(('event_%s' % ii, ii)
+                              for ii in unique_event_types)
+        ev_dict_rev = dict((value, key) for key, value in event_dict.items())
+
+        # Turn event_ixs from seconds to indices in case sfreq != 1
+        event_ixs = (event_ixs * self.sfreq).astype(int)
+
+        # Iterate through event types and create columns of event onsets.
+        events_continuous = np.zeros([self.n_times,
+                                      len(unique_event_types) + n_covs])
+        event_names = []
+        for ii, ev_type in enumerate(unique_event_types):
+            msk_events = event_ids == ev_type
+            i_ev = event_ixs[msk_events]
+            events_continuous[i_ev, ii] = 1
+
+            # Handle event names
+            event_names.append(ev_dict_rev[ev_type])
+        if n_covs > 0:
+            for iev, icov in zip(event_ixs, covs):
+                events_continuous[iev, -n_covs:] = icov
+            event_names = event_names + cov_names
+            events_continuous = sparse.csr_matrix(events_continuous)
+
+        self.names_ = event_names
+        self.unique_event_types_ = unique_event_types
+        return events_continuous
+
+    def fit_transform(self, event_ixs, event_ids=None, event_dict=None,
+                      covariates=None, covariate_names=None):
+        """Does nothing, only used for sklearn compatibility.
+
+        Parameters
+        ----------
+        X : array
+        y : array
+
+        Returns
+        -------
+        self
+        """
+        out = self.transform(
+            event_ixs, event_ids=event_ids, event_dict=event_dict,
+            covariates=covariates, covariate_names=covariate_names)
+        return out
+
+
+def delay_timeseries(ts, delays, sfreq=1.):
+    """Return a time-lagged input timeseries.
+
+    Parameters
+    ----------
+    ts: array, shape (n_times, n_feats)
+        The timeseries to delay
+    delays: array of floats, shape (n_delays) |
+            list of arrays, len == n_feats
+        The time (in seconds) of each delay for specifying
+        pre-defined delays. Negative means timepoints in the past,
+        positive means timepoints in the future. If a list, then
+        one may supply a different number of delays per feature.
+    sfreq: int
+        The sampling frequency of the series
+
+    Returns
+    -------
+    delayed: array, shape(n_times, n_feats * n_delays)
```

Comment:
I'm wondering whether the output should be forced to 2D like this. I guess this is because of the usecases jona described us (i.e. different lags depending on X).

Can you add an XXX to remember that we may want to add a 'vectorizer=True' param?